### PR TITLE
dont try to copy trash items to user if the user has not enough free space

### PR DESCRIPTION
**Pull request by icewind1991**
If it tries to copy anyway a `CopyRecursiveException` will be thrown later

### CODE REVIEW

File: apps/files_trashbin/lib/Trashbin.php [hunk 1]
```diff
@@ -189,7 +189,9 @@ private static function copyFilesToUser($sourcePath, $owner, $targetPath, $user,
 
 		$target = $user . '/files_trashbin/files/' . $targetFilename . '.d' . $timestamp;
 		$source = $owner . '/files_trashbin/files/' . $sourceFilename . '.d' . $timestamp;
-		self::copy_recursive($source, $target, $view);
+		if ($view->free_space($target) > 0) {
```
Comment (MorrisJobke) [inline apps/files_trashbin/lib/Trashbin.php]: And what if the source is bigger than the free space?

Comment (icewind1991) [reply apps/files_trashbin/lib/Trashbin.php]: changed to check agains the size of the trash item

File: apps/files_trashbin/lib/Trashbin.php [hunk 2]
```diff
@@ -189,7 +189,10 @@ private static function copyFilesToUser($sourcePath, $owner, $targetPath, $user,
 
 		$target = $user . '/files_trashbin/files/' . $targetFilename . '.d' . $timestamp;
 		$source = $owner . '/files_trashbin/files/' . $sourceFilename . '.d' . $timestamp;
-		self::copy_recursive($source, $target, $view);
+		$free = $view->free_space($target);
+		if (($free < 0) || ($view->filesize($source) < $free)) {
```
Comment (MorrisJobke) [inline apps/files_trashbin/lib/Trashbin.php]: Shouldn't it be 
```suggestion
		if (($free > 0) || ($view->filesize($source) < $free)) {
```
?

Comment (icewind1991) [reply apps/files_trashbin/lib/Trashbin.php]: No, when free space is negative it should always (try to) copy

negative free space values mean unknown/unlimited free space

Comment (MorrisJobke) [reply apps/files_trashbin/lib/Trashbin.php]: Ah ... got it 👍

Comment (MorrisJobke) [inline apps/files_trashbin/lib/Trashbin.php]: Then I would propose a more readable `if` statement:

```suggestion
		$isUnknownOrUnlimitedFreeSpace = $free < 0;
		$isEnoughFreeSpaceLeft = $view->filesize($source) < $free;
		if ($isUnknownOrUnlimitedFreeSpace || $isEnoughFreeSpaceLeft) {
```

Comment (MorrisJobke) [reply apps/files_trashbin/lib/Trashbin.php]: Then I would propose a more readable `if` statement:

```suggestion
		$isUnknownOrUnlimitedFreeSpace = $free < 0;
		$isEnoughFreeSpaceLeft = $view->filesize($source) < $free;
		if ($isUnknownOrUnlimitedFreeSpace || $isEnoughFreeSpaceLeft) {
```

**Review by gary-kim — APPROVED** [review]
Improving readability would be nice as @MorrisJobke suggested but other than that, LGTM.

**Review by rullzer — APPROVED** [review]

**Pull request merged**